server/stream.go

```diff
@@ -4716,14 +4730,24 @@ func (mset *stream) Store() StreamStore {
 
 // Determines if the new proposed partition is unique amongst all consumers.
 // Lock should be held.
-func (mset *stream) partitionUnique(partition string) bool {
-	for _, o := range mset.consumers {
-		if o.cfg.FilterSubject == _EMPTY_ {
-			return false
-		}
-		if subjectIsSubsetMatch(partition, o.cfg.FilterSubject) ||
-			subjectIsSubsetMatch(o.cfg.FilterSubject, partition) {
-			return false
+func (mset *stream) partitionUnique(partitions []string) bool {
+	for _, partition := range partitions {
+		for _, o := range mset.consumers {
+
+			// if multiple filters have _EMPTY_ filter, it can be the only one, otherwise
+			// it would have overlapping filters.
+			multipleFiltersWithEmpty := len(o.cfg.FilterSubjects) == 1 && o.cfg.FilterSubjects[0] == _EMPTY_
+
+			if o.cfg.FilterSubject == _EMPTY_ || multipleFiltersWithEmpty {
+				return false
+			}
+			for _, subject := range subjectFilters(o.cfg.FilterSubject, o.cfg.FilterSubjects) {
```

Comment:
Should subjectFilters be a method on o? o.subjectFilters()?

Reply:
Was thinking about it, but I use this function before creating `consumer` to do the early validation.

I plan to refactor Consumer validation and initialization, but it would clutter this PR and make it hard to read.
Then, it definately could be a method.